src/libYARP_sig/src/yarp/sig/Image.cpp

```diff
@@ -875,6 +875,36 @@ bool Image::copy(const Image& alt)
 }
 
 
+bool Image::move(Image&& alt) noexcept
+{
+    // Cannot move an image of the wrong type inside an ImageOf that does not
+    // support it.
+    yAssert(dynamic_cast<FlexImage*>(this) ||
+            getPixelCode() == alt.getPixelCode());
+    if (&alt != this) {
+        delete static_cast<ImageStorage*>(implementation);
+        implementation = std::exchange(alt.implementation, nullptr);
+        synchronize();
+    }
+    return true;
+}
+
+
+bool Image::swap(Image& alt)
+{
+    // Cannot swap two ImageOf of different type, or an image of the wrong type
+    // inside an ImageOf that does not support it.
+    yAssert((dynamic_cast<FlexImage*>(this) && dynamic_cast<FlexImage*>(&alt)) ||
+            getPixelCode() == alt.getPixelCode());
+    if (&alt != this) {
+        std::swap(alt.implementation, implementation);
+        synchronize();
+        alt.synchronize();
+    }
+    return true;
+}
+
+
```

Comment:
What about using this `move` and `swap` for refactoring the moves operators?

Reply:
It can be done, but I think it will make the move slower (at least in debug, since a dynamic_cast is required to check the type, which you don't need).
I think they could just be simplified instead...
The move constructor could just become
```diff
 Image::Image(Image&& other) noexcept
-    : implementation(other.implementation)
+    : implementation(std::exchange(other.implementation, nullptr))
 {
-    other.implementation = nullptr;
     synchronize();
 }
```

and the move assignment operator

```diff
 Image& Image::operator=(Image&& other) noexcept
 {
-    Image moved(std::move(other));
-    std::swap(moved.implementation, implementation);
+    implementation = std::exchange(other.implementation, nullptr);
     synchronize();
     return *this;
 }
```